server/src/main/java/org/apache/druid/server/coordinator/DruidCoordinatorConfig.java

```diff
@@ -43,9 +45,16 @@ public abstract class DruidCoordinatorConfig
   @Default("PT1H")
   public abstract Duration getCoordinatorMetadataStoreManagementPeriod();
 
+  @Config("druid.coordinator.kill.on")
+  @Default("false")
+  public abstract Boolean getAutoKillEnabled();
+
+  @Nullable
   @Config("druid.coordinator.kill.period")
-  @Default("P1D")
-  public abstract Duration getCoordinatorKillPeriod();
+  public Duration getCoordinatorKillPeriod()
+  {
+    return null;
```

Comment:
```suggestion
    return getCoordinatorIndexingPeriod();
```

This should simplify some of the null handling added elsewhere and makes it clear that the default is the indexing period